test/built-ins/String/prototype/replaceAll/searchValue-replacer-RegExp-call.js

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2019 Leo Balter. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-string.prototype.replaceall
+description: >
+  A RegExp searchValue's Symbol.replace can be called instead of the next steps of replaceAll
+info: |
+  String.prototype.replaceAll ( searchValue, replaceValue )
+
+  1. Let O be RequireObjectCoercible(this value).
+  2. If searchValue is neither undefined nor null, then
+    a. Let isRegExp be ? IsRegExp(searchString).
+    b. If isRegExp is true, then
+      i. Let flags be ? Get(searchValue, "flags").
+      ii. Perform ? RequireObjectCoercible(flags).
+      iii. If ? ToString(flags) does not contain "g", throw a TypeError exception.
+    c. Let replacer be ? GetMethod(searchValue, @@replace).
+    d. If replacer is not undefined, then
+      i. Return ? Call(replacer, searchValue, « O, replaceValue »).
+  3. Let string be ? ToString(O).
+  4. Let searchString be ? ToString(searchValue).
+  ...
+features: [String.prototype.replaceAll, Symbol.replace, class]
+---*/
+
+let called = 0;
+
+class RE extends RegExp {
+  [Symbol.replace](...args) {
+    const actual = super[Symbol.replace](...args);
+
+    // Ordering is intentional to observe call from super
+    called += 1;
+    return actual;
+  }
+
+  toString() {
+    throw 'Should not call toString on searchValue';
+  }
+}
+
+const samples = [
```

Comment:
Wow, this test is intense—even with `replace` itself, neither JSC nor SM pass it. I guess that's fine though, if the `replace` version is coming in a future PR.

Reply:
I believe this is much more important to RegExp.prototype[Symbol.replace], which is the part we are really testing here. So my plan is in this priority order:

1. Extend tests coverage for `RegExp.prototype[Symbol.replace]`
1. Create a feature flag for `RegExp.prototype[Symbol.replace]` and `String.prototype.replace`. Not new features, but there are too much tangliness here. _Yes, I invented a new word_.
1. Extend tests coverage for `String.prototype.replace`
1. Remove redundant legacy tests for `String#replace` we will eventually find as duplicates of the new coverage from the previous item.

That won't hurt anyone and will only add coverage. If the tests are failing on 2 engines, I believe this points to a bug withing the RegExp method, which we are missing to flag in the appropriate place.

I also